src/Actions.ts

```diff
@@ -18,19 +19,79 @@ export class Actions {
         // These are called in server.ts without scoping
         this.onButtonAction = this.onButtonAction.bind(this);
         this.onStaticSelectAction = this.onStaticSelectAction.bind(this);
+        this.onModalAction = this.onModalAction.bind(this);
         this.createPollRoute = this.createPollRoute.bind(this);
+        this.submitModal = this.submitModal.bind(this);
     }
 
-    public postMessage(channel: string, text: string, blocks: KnownBlock[], user?: string): Promise<WebAPICallResult> {
+    public postMessage(channel: string, text: string, blocks: KnownBlock[]): Promise<WebAPICallResult> {
         const msg: ChatPostMessageArguments = { channel, text, blocks };
-        if (user) {
-            msg.user = user;
-        } else {
-            msg.as_user = false;
-        }
         return this.wc.chat.postMessage(msg);
     }
 
+    public async displayModal(channelId: string, triggerId: string): Promise<void> {
+        const modal = new PollModal(channelId);
+        const response = await this.wc.views.open({
+            trigger_id: triggerId,
+            view: modal.constructModalView(),
+        });
+        ModalMap.set((response as any).view.id, modal);
+        return;
+    }
+
+    public onModalAction(payload: any, res: (message: any) => Promise<unknown>): {text: string} {
+        const currentModal = ModalMap.get(payload.view.id);
+        const actionId = payload.actions[0].action_id;
+        // We don't do anything if viewID is invalid
+        if (!currentModal) return { text: "Modal not found!" };
+        if (actionId === "add_option") {
+            currentModal.addOption();
+            this.wc.views.update({
+                view_id: payload.view.id,
+                view: currentModal.constructModalView(),
+            });
+        }
+        return { text: "Action processed" };
```

Comment:
Maybe make this more explicit so the verbosity actually means something

Reply:
The user doesn't see this, it's just to pacify Slack's API. I could return an empty string, but I'll specify that it's a modal action.